src/main/webapp/app/exam/participate/summary/exam-participation-summary.component.ts

```diff
@@ -27,9 +29,40 @@ export class ExamParticipationSummaryComponent implements OnInit {
     readonly IncludedInOverallScore = IncludedInOverallScore;
     readonly SUBMISSION_TYPE_ILLEGAL = SubmissionType.ILLEGAL;
 
+    /**
+     * If true, expects {@link studentExam} input is provided before ngOnInit and fetches StudentExamWithGradeDTO
+     * (without studentExam) from server itself.
+     * Otherwise, expects {@link studentExamWithGradeDTO} input is provided by the parent component with {@link StudentExamWithGradeDTO#studentExam} set.
+     */
+    @Input()
+    shouldFetchGradeInfoSeparately = false;
+
+    /**
+     * Current student's exam.
+     * @see shouldFetchGradeInfoSeparately
+     */
     @Input()
     studentExam: StudentExam;
 
+    /**
+     * Sets {@link studentExamWithGradeDTO} and also {@link studentExam} if {@link StudentExamWithGradeDTO#studentExam} is set.
+     * Allows doing one less HTTP request if {@link studentExamWithGradeDTO} is provided with {@link StudentExamWithGradeDTO#studentExam}
+     * and {@link shouldFetchGradeInfoSeparately} is false.
+     *
+     * @see shouldFetchGradeInfoSeparately
+     *
+     * @param studentExamWithGrade Grade info for current student's exam, possibly containing {@link StudentExamWithGradeDTO#studentExam}.
+     */
+    @Input()
+    set studentExamWithGrade(studentExamWithGrade: StudentExamWithGradeDTO) {
```

Comment:
why do we need this setter as separate input?

Reply:
Converted it to a non-input field, now `studentExam` is taken as input and grade info is fetched from server.